component/mitz/component.go

```
@@ -0,0 +1,549 @@
+	gatewaySystem        string
+	sourceSystem         string
+	endpointQuerier      EndpointQuerier
+}
+
+const (
+	subscriptionPath = "/abonnementen/fhir"
+	consentCheckPath = "/geslotenautorisatievraag/xacml3"
+)
+
+// New creates a new MITZ component
+func New(config Config, endpointQuerier EndpointQuerier) (*Component, error) {
+	if config.MitzBase == "" {
+		return nil, fmt.Errorf("mitzbase must be configured when MITZ component is enabled")
+	}
+
+	// Parse base URL and construct subscription endpoint
+	subscriptionURL, err := url.Parse(config.MitzBase + subscriptionPath)
+	if err != nil {
+		return nil, fmt.Errorf("invalid subscription endpoint: %w", err)
+	}
+
+	// Create HTTP client with optional mTLS configuration
+	httpClient, err := createHTTPClient(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP client: %w", err)
+	}
+
+	consentCheckEndpoint := config.MitzBase + consentCheckPath
+
+	return &Component{
+		client:               fhirclient.New(subscriptionURL, httpClient, fhirutil.ClientConfig()),
+		httpClient:           httpClient,
+		consentCheckEndpoint: consentCheckEndpoint,
+		gatewaySystem:        config.GatewaySystem,
+		sourceSystem:         config.SourceSystem,
+		endpointQuerier:      endpointQuerier,
+	}, nil
+}
+
+// createHTTPClient creates an HTTP client with optional mTLS configuration
+func createHTTPClient(config Config) (*http.Client, error) {
+	client := &http.Client{
+		Transport: http.DefaultTransport,
+	}
+
+	// If TLS certificate is configured, set up mTLS
+	if config.TLSCertFile != "" {
+		tlsConfig, err := tlsutil.CreateTLSConfig(tlsutil.Config{
+			CertFile: config.TLSCertFile,
+			KeyFile:  config.TLSKeyFile,
+			Password: config.TLSKeyPassword,
+			CAFile:   config.TLSCAFile,
+		})
+		if err != nil {
+			// Log warning but don't fail - allow running without mTLS
+			log.Warn().Err(err).Str("certFile", config.TLSCertFile).Msg("Failed to load TLS certificate, continuing without mTLS")
```

If the user configured mTLS, but the setup throws an error we probably want the code to fail early. We allow running without TLS, but only when explicitly configured to do so I'd say